sites/target.py

```diff
@@ -0,0 +1,199 @@
+            monitor_delay), float(error_delay)
+
+        starting_msg = "Starting Target"
+        self.browser = self.init_driver()
+        self.product_image = None
+
+        if settings.dont_buy:
+            starting_msg = "Starting Target in dev mode; will not actually checkout"
+
+        self.status_signal.emit(self.create_msg(starting_msg, "normal"))
+        self.status_signal.emit(self.create_msg("Logging In..", "normal"))
+        self.login()
+        self.monitor()
+        self.atc()
+        self.checkout()
+        self.submit_billing()
+
+        if not settings.dont_buy:
+            self.submit_order()
+        else:
+            self.status_signal.emit(self.create_msg("Mock Order Placed", "success"))
+            send_webhook("OP", "Target", self.profile["profile_name"], self.task_id, self.product_image)
+
+    # https://stackoverflow.com/questions/33225947/can-a-website-detect-when-you-are-using-selenium-with-chromedriver
+    def change_driver(self, loc):
+        print(loc)
+        fin = open(loc, 'rb')
+        data = fin.read()
+        val = "$cdc_" + "".join(random.choices(string.ascii_letters + string.digits, k=22)) + "_"
+
+        result = re.search(b"[$]cdc_[a-zA-Z0-9]{22}_", data)
+
+        if result is not None:
+            self.status_signal.emit(self.create_msg("Changing value in Chromedriver", "normal"))
+            data = data.replace(result.group(0), val.encode())
+            fin.close()
+            fin = open(loc, 'wb')
+            fin.truncate()
+            fin.write(data)
+            fin.close()
+        else:
+            fin.close()
+
+    def init_driver(self):
+        driver_manager = ChromeDriverManager()
+        driver_manager.install()
+        self.change_driver(driver_path)
+        browser = webdriver.Chrome(driver_path)
+
+        browser.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
+            "source": """
+                  Object.defineProperty(navigator, 'webdriver', {
+                   get: () => undefined
+                  })
+                """
+        })
+
+        return browser
+
+    def login(self):
+        self.browser.get("https://www.target.com")
+
+        self.browser.find_element_by_id("account").click()
+
+        wait(self.browser, 3).until(EC.element_to_be_clickable((By.ID, "accountNav-signIn"))).click()
+
+        my_username = "username"
+        my_password = "password"
+
+        wait(self.browser, 3).until(EC.presence_of_element_located((By.ID, "username"))).send_keys(my_username)
+        password = self.browser.find_element_by_id("password")
+        password.send_keys(my_password)
+        self.browser.find_element_by_id("login").click()
+
+        # Gives it time for the login to complete
+        time.sleep(random_delay(self.monitor_delay, settings.rand_delay_start, settings.rand_delay_stop))
```

Comment:
In PR #30  I renamed settings.rand_delay_start and settings.rand_delay_stop to settings.random_delay_start and settings.random_delay_start

Reply:
Renamed